### ISSUE TITLE
Design a strategy for login token invalidation

### ISSUE THREAD

**Issue by iamareebjamal** [fossasia/open-event-orga-app#113]
Currently, the login token is cached forever. In actuality, it expires after some time.

Either determine from server the time it takes for a token to expire and delete token at that time, or when it does, instead of showing an error message, trigger the presenter to load the login screen again

What should be a better approach @niranjan94 @mejariamol

**Comment by niranjan94** [comment]
The token is a JW(T)oken .. You will be able to parse the token directly to get it's expiry time.

**Comment by niranjan94** [comment]
https://jwt.io/

**Comment by niranjan94** [comment]
This is a sample token form the organizer server for example in the format 

```
<header>.<payload>.<signature>
```

```
eyJhbGciOiJIUzs2dssIsInR5cCI6IkpXVCJ9.eyJpZGVudGl0eSI6MTEyLCJpYXQiOjE0OTUzODQ3NDMsIm5iZiI6MTQ5NTM4NDc0MywiZXhwIjoxNDk1NDcxMTQzfQ.c2P58CK9vE_vbsdLcfIn1mS-szEGFK-LAhxK1Q5YT10
```

Here the payload is 

```
eyJpZGVudGl0eSI6MTEyLCJpYXQiOjE0OTUzODQ3NDMsIm5iZiI6MTQ5NTM4NDc0MywiZXhwIjoxNDk1NDcxMTQzfQ
```

Base64 decoding the payload gives us 

```js
{"identity":112,"iat":1495384743,"nbf":1495384743,"exp":1495471143}
```

**Comment by iamareebjamal** [comment]
@niranjan94 Great, will implement the expiry time invalidation

**Comment by mejariamol** [comment]
@niranjan94 @iamareebjamal why do we need to implement token expiry? why can't just get the token once and save locally forever unless user manually log out?

**Comment by niranjan94** [comment]
Because @mejariamol , irrespective of whether the user logs out or not, the token will expire at the given time. You will no longer be able to make requests with that token once it expires

**Comment by mejariamol** [comment]
@iamareebjamal 
>Either determine from server the time it takes for a token to expire and delete token at that time, or when it does, instead of showing an error message, trigger the presenter to load the login screen again

instead triggering the presenter to load the login screen again, let the app itself refresh token for user. Are we storing username and password ?

@niranjan94 what is your view?

**Comment by niranjan94** [comment]
@mejariamol the tokens are now valid for 24 hours. I think it's acceptable for the user to login every 24 hours.

**Comment by mejariamol** [comment]
@iamareebjamal are you working on this already?

**Comment by iamareebjamal** [comment]
Yeah, sorry, exams and all. If you want to take it up, please do

**Comment by iamareebjamal** [comment]
Working on it

**Comment by iamareebjamal** [comment]
Fixed by #132
